src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, hide) {
+    super(name);
+  }
+
+  bite(victim) {
+    const animal = Animal.alive.find(elem => elem === victim);
+
+    if (animal.hidden !== true && !(animal instanceof Carnivore)) {
+      animal.health -= 50;
+    }
+
+    if (animal.health <= 0) {
+      Animal.alive.splice([Animal.alive.indexOf(animal)], 1);
+    }
```

Comment:
you could just minus health of victim and then filter Animal (health > 0) instead of such logic